modules/google-analytics/classes/wp-google-analytics-legacy.php

```diff
@@ -123,11 +136,43 @@ public function insert_code() {
 					ga.src = ('https:' === document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
 					var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
 				})();
-			</script>\r\n",
+			</script>
+			<!-- End Jetpack Google Analytics -->\r\n",
 			implode( "\r\n", $custom_vars )
 		);
 	}
 
+	/**
+	 * Renders new gtag code.
+	 *
+	 * @param string $tracking_id Google Analytics measurement ID.
+	 */
+	private function render_gtag_code( $tracking_id ) {
+		/**
+		 * Allow for additional elements to be added to the Global Site Tags array.
+		 *
+		 * @since 9.2.0
+		 *
+		 * @param array $universal_commands Array of gtag function calls.
+		 */
+		$universal_commands = apply_filters( 'jetpack_gtag_universal_commands', array() );
+
+		// phpcs:disable WordPress.WP.EnqueuedResources.NonEnqueuedScript
+		?>
+		<!-- Jetpack Google Analytics -->
+		<script async src='https://www.googletagmanager.com/gtag/js?id=<?php echo esc_attr( $tracking_id ); ?>'></script>
+		<script>
+			window.dataLayer = window.dataLayer || [];
+			function gtag() { dataLayer.push( arguments ); }
+			gtag( 'js', new Date() );
+			gtag( 'config', '<?php echo esc_js( $tracking_id ); ?>' );
```

Comment:
Is `esc_js` correct here? While it seems unlikely, if `$tracking_id` contains `&`, `"`, `<`, or `>` those will be HTML entity-encoded rather than encoded for JavaScript.

It seems to me that `json_encode()` would work better.

Reply:
Done

Reply:
I tried the recommendations and I get this:

<img width="380" alt="Screen Shot 2020-11-19 at 11 45 31 PM" src="https://user-images.githubusercontent.com/66652282/99761594-7b912b00-2ac4-11eb-82da-d0ebebcf731e.png">

Reverting the suggestions work as expected.

Reply:
Sorry, I should have explained better.
```suggestion
			gtag( 'config', <?php echo wp_json_encode( $tracking_id ); ?> );
```
is what I meant. Since json_encode already quotes strings, you don't have to include quotes of your own.

Reply:
Ah, thanks @anomiex! Didn't notice that.